app/controllers/idv/image_uploads_controller.rb

```diff
@@ -61,6 +57,25 @@ def doc_auth_client
       @doc_auth_client ||= DocAuthRouter.client
     end
 
+    def update_analytics(client_response)
+      add_costs(client_response)
+      update_funnel(client_response)
+      analytics.track_event(
+        Analytics::IDV_DOC_AUTH_SUBMITTED_IMAGE_UPLOAD_VENDOR,
+        client_response.to_h.merge(user_id: image_form.document_capture_session.user.uuid),
+      )
+    end
+
+    def update_funnel(result)
+      user_id = image_form.document_capture_session.user.id
+      issuer = sp_session[:issuer]
+      steps = %i[front_image back_image]
+      steps << :selfie if liveness_checking_enabled?
+      steps.each do |step|
+        Funnel::DocAuth::RegisterStep.new(user_id, issuer).call(step.to_s, :update, result.success?)
```

Comment:
This won't capture failures from `form_response.success?` being false in the `create` method. Is that a problem?

Reply:
Yes it's ok.  For overall success we want to limit it to those users that submitted a legit image at least once.  The form looks like it catches invalid files and throttles.